### DISCUSSION TITLE
Callback run twice when should only once

### DISCUSSION THREAD

**Discussion by Hootie9lives** [Coding-with-Adam/Dash-by-Plotly#8] [Q&A]
Hi @Coding-with-Adam,

I am curious if there is any insights you can share with me on this issue I'm having. 

    @app.callback(
        Output({'type': 'jira_modal', 'index': MATCH}, "is_open"),
        Output({'type': 'modal_form', 'index': MATCH}, "children"),
        Output({'type': 'modal_form', 'index': MATCH}, "style"),
        Output({'type': 'id', 'index': MATCH}, 'data'),
        Input({'type': 'decision', 'index': MATCH}, 'value'),
        Input("modal_submit", "n_clicks"),
        State({'type': 'name', 'index': MATCH}, 'value'),
        State({'type': 'country', 'index': MATCH}, 'value'),
        State({'type': 'jira_modal', 'index': MATCH}, "is_open"),
        State("session", "data"),
        State("user_id", "data"), prevent_initial_call=True)

I have a callback like this. Whenever I click on modal_submit button to change the Is_open state of the modal, the callback is ran twice.

**Comment by Coding-with-Adam** [comment]
Sorry, it's hard to say why without seeing the whole code, @Hootie9lives 
I'm not sure what's happening there.
